JustSaying.AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -146,57 +146,38 @@ public void StopListening()
             var region = _queue.Region.SystemName;
+            ReceiveMessageResponse sqsMessageResponse = null;
 
             try
             {
-                var numberOfMessagesToReadFromSqs = await GetNumberOfMessagesToReadFromSqs()
-                    .ConfigureAwait(false);
-
-                var watch = System.Diagnostics.Stopwatch.StartNew();
-
-                var request = new ReceiveMessageRequest
-                    {
-                        QueueUrl = _queue.Url,
-                        MaxNumberOfMessages = numberOfMessagesToReadFromSqs,
-                        WaitTimeSeconds = 20
-                    };
-
-                var receiveTimeout = new CancellationTokenSource(TimeSpan.FromSeconds(30));
-                ReceiveMessageResponse sqsMessageResponse;
-                try
-                {
-                    using (var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(ct, receiveTimeout.Token))
-                    {
-                        sqsMessageResponse = await _queue.Client.ReceiveMessageAsync(request, linkedCts.Token)
-                            .ConfigureAwait(false);
-                    }
-                }
-                finally
-                {
-                    if (receiveTimeout.Token.IsCancellationRequested)
-                    {
-                        Log.Info("Receiving messages from queue {0}, region {1}, timed out", queueName, region);
-                    }
-                }
-
-                watch.Stop();
-
-                _messagingMonitor.ReceiveMessageTime(watch.ElapsedMilliseconds, queueName, region);
-
+                sqsMessageResponse = await GetMessagesFromSqsQueue(ct, queueName, region);
                 var messageCount = sqsMessageResponse.Messages.Count;
 
                 Log.Trace(
                     $"Polled for messages - Queue: {queueName}, Region: {region}, MessageCount: {messageCount}");
-
-                foreach (var message in sqsMessageResponse.Messages)
-                {
-                    HandleMessage(message);
-                }
             }
             catch (InvalidOperationException ex)
             {
                 Log.Trace(
-                    $"Suspected no message in queue [{queueName}], region: [{region}]. Ex: {ex}");
+                    $"Could not determine number of messages to read from [{queueName}], region: [{region}]. Ex: {ex}");
+            }
+            catch (OperationCanceledException ex)
+            {
+                Log.Trace($"Suspected no message in queue [{queueName}], region: [{region}]. Ex: {ex}");
+            }
+            catch (Exception ex)
+            {
+                var msg = $"Issue receiving messages for queue {queueName}, region {region}";
+                Log.Error(ex, msg);
+            }
+
+            try
+            {
+                if (sqsMessageResponse != null)
+                    foreach (var message in sqsMessageResponse.Messages)
```

Comment:
@AnthonySteele suprised you didn't comment on the lack of `{` & `}` 😄
I think this PR is good to merge.